packages/core/src/types.ts

```diff
@@ -1279,30 +1279,39 @@ export type ExtractEvent<
   TEventType extends TEvent['type']
 > = TEvent extends { type: TEventType } ? TEvent : never;
 
-export interface ActorRef<TEvent extends EventObject, TEmitted = any>
-  extends Subscribable<TEmitted> {
+export interface BaseActorRef<TEvent extends EventObject, TEmitted = any> {
   send: Sender<TEvent>;
+  subscribe?: Subscribable<TEmitted>['subscribe'];
 }
 
-export interface SpawnedActorRef<TEvent extends EventObject, TEmitted = any>
-  extends ActorRef<TEvent, TEmitted> {
+export interface ActorRef<TEvent extends EventObject, TEmitted = any>
+  extends Subscribable<TEmitted> {
+  send: Sender<TEvent>;
   id: string;
   getSnapshot: () => TEmitted | undefined;
   stop?: () => void;
```

Comment:
q for v5: should we kill `.stop` on ActorRefs? should such direct call only be allowed for parents? being able to stop an arbitrary actor when the ref gets shared seems to be dangerous

Reply:
You should be able to stop an actor from any other actor that has a ref to it. This matches the behavior in Akka; I don't see how this can be dangerous (obviously, it's the responsibility of the developer to stop it properly): https://alvinalexander.com/scala/how-to-stop-akka-actors-in-scala-gracefulstop-poisonpill/

Reply:
IMHO there are two dangers here:
1. it's an imperative/eager call, we should prefer pure actions 
2. that actor may live both in contexts of different actors and also in internal structures of its parent, especially for the latter we can't just shift the responsibility of handling that to the user as they don't control that at all

According to the Akka docs we can find [this](https://doc.akka.io/docs/akka/current/typed/from-classic.html#stopping):
> Classic actors can be stopped with the stop method of ActorContext or ActorSystem. [...] There is also a stop method in the ActorContext but it can only be used for stopping direct child actors and not any arbitrary actor.
and [this](https://doc.akka.io/docs/akka/current/actors.html#stopping-actors):
> Typically the context is used for stopping the actor itself or child actors and the system for stopping top level actors

So it seems to me that stopping the arbitrary actor might not be a recommended pattern in Akka, even if possible.

Reply:
You're right, but let's discuss in a separate PR, as this might be a breaking change.

Reply:
Card added: https://github.com/davidkpiano/xstate/projects/1#card-62575983